### PR TITLE
Use `baddbmm` to reduce the number of kernel calls when running T5

### PR DESCRIPTION
**Pull request by thomasw21**
# What does this PR do?

Reduce the number of kernel calls by using `baddbmm` and built-in `F.softmax(..., dtype=torch.float))`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by thomasw21** [comment]
Arf turns out this doesn't work without having extra copies. Closing as I'm not sure if it's worth it.
